datalad_next/iter_collections/utils.py

```diff
@@ -28,9 +29,26 @@ class FileSystemItemType(Enum):
     specialfile = 'specialfile'
 
 
-@dataclass  # sadly PY3.10+ only (kw_only=True)
-class FileSystemItem:
+@dataclass
+class NamedItem:
+    name: Any
+
+
+@dataclass
+class TypedItem:
+    type: Any
+
+
+@dataclass
+class PathBasedItem(NamedItem):
+    # a path-identifier in an appropriate context.
+    # could be a filename, a relpath, or an absolute path.
+    # should match platform conventions
```

Comment:
Would an item in a tar-file or zip-file be a "PathBasedItem" or only a "NamedItem"? In the former case, the "platform" from the comment "should match platform conventions" would refer to tar-file or zip-file spec. Which is OK, I am just mentioning it.

Reply:
I think it would be `PathBasedItem`.

I have a comment somewhere that is wondering whether there are archive formats that use non-POSIX member names. Do you know any?

Reply:
Ah, now I see what you mean. No, the "platform" is not tar-file or zip-file. The question is posix path of windows path. However, given that this is nothing specific to this PR. I have filed https://github.com/datalad/datalad-next/issues/402 so we can move on here.